plugins/lookup/get_moid.py

```diff
@@ -0,0 +1,579 @@
+            _url = self.build_url('host', _filter)
+            _result = await self.fetch(_url)
+            if _result and len(_result) == 1:
+                if objects_list and _result[0]['name'] == objects_list[0]:
+                    del objects_list[0]
+                try:
+                    host_moid = _result[0]['host']
+                except (TypeError, KeyError, IndexError) as e:
+                    raise AnsibleError(to_native(e))
+            elif _result and len(_result) > 1:
+                raise AnsibleError("More than one object available: [%s]."
+                                   % ", ".join(list(item['name'] for item in _result)))
+            else:
+                return None
+
+            if object_type == 'host':
+                return host_moid
+
+            # look for an object inside a host
+            _filter = {}
+            if object_name:
+                _filter["names"] = object_name
+            _filter["hosts"] = host_moid
+            _url = self.build_url(object_type, _filter)
+            _result = await self.fetch(_url)
+            if _result and len(_result) == 1:
+                try:
+                    obj_moid = _result[0][object_type]
+                except (TypeError, KeyError, IndexError) as e:
+                    raise AnsibleError(to_native(e))
+                return obj_moid
+            elif _result and len(_result) > 1:
+                raise AnsibleError("More than one object available: [%s]."
+                                   % ", ".join(list(item['name'] for item in _result)))
+            else:
+                return None
+
+        # If no VM has been found, it might be in a resource_pool
+        # GET resource_pools MoID
+        _result = await self.get_folder_or_rp_moid(dc_moid, objects_list, 'resource_pool',
+                                                   cluster_moid=cluster_moid)
+        if _result:
+            try:
+                rp_moid = _result[-1]['resource_pool']
+            except (TypeError, KeyError, IndexError) as e:
+                raise AnsibleError(to_native(e))
+        else:
+            return []
+
+        # search for VM
+        if object_type == 'vm':
+            if object_name:
+                _filter["names"] = object_name
+            _filter["resource_pools"] = rp_moid
+            _filter["clusters"] = cluster_moid
+            _url = self.build_url(object_type, _filter)
+            _result = await self.fetch(_url)
+            if _result:
+                try:
+                    vm_moid = _result[0]['vm']
+                except (TypeError, KeyError, IndexError) as e:
+                    raise AnsibleError(to_native(e))
+                return vm_moid
+            else:
+                return None
+
+        return rp_moid
+
+    async def helper_moid(self, object, parent_object, object_type, _filter):
+        result = None
+        parent_key = 'parent_' + object_type + 's'
```

Comment:
Since we've focusing on Py36+, you can use a fancy f-string instead :-).
`parent_key = f'parent_{object_type}s'`